framework/src/problems/FEProblemBase.C

```diff
@@ -2929,6 +2929,12 @@ FEProblemBase::addFVInterfaceKernel(const std::string & fv_ik_name,
                                     const std::string & name,
                                     InputParameters & parameters)
 {
+  const auto nl_sys_num =
+      determineNonlinearSystem(parameters.varName("variable1", name), true).first
+          ? determineNonlinearSystem(parameters.varName("variable1", name), true).second
+          : (unsigned int)0;
+  parameters.set<SystemBase *>("_sys") = _nl[nl_sys_num].get();
```

Comment:
I don't know if I love this or the modification in the add-objects-helper-routine. An alternative would be to not use the `_sys` parameter in the `FVInterfaceKernel` object at all and just `_var1.sys()` and `_var2.sys()`.

The modification here doesn't consider the possibility of a displaced mesh which is going to be a real possibility after https://github.com/idaholab/moose/pull/25181